[BUG] Incorrect casing in generated Prisma types

## Information

- **Version:** 5.x
- Packages: @tsed/prisma

When using lowercase model names in Prisma, the repository types generated by @tsed/prisma have incorrect casing on the `aggregate` parameter which results in broken types.
Prisma's behavior is likely caused by this line:
https://github.com/prisma/prisma/blob/f35af40ffed4cc96d990ab26c96d99c5f104c178/packages/client/src/generation/utils.ts#L134

## Example

```
model user {
  id            String
}
```

```
#13 12.77 node_modules/.prisma/client/entities/repositories/userRepository.d.ts(18,28): error TS2724: '"/app/node_modules/.prisma/client/index".Prisma' has no exported member named 'userAggregateArgs'. Did you mean 'UserAggregateArgs'?
```

## Acceptance criteria

- [ ] @tsed/prisma should generate valid types

--- a/packages/orm/prisma/package.json
+++ b/packages/orm/prisma/package.json
@@ -16,12 +16,10 @@
     "tsed-prisma": "lib/cjs/generator.js"
   },
   "scripts": {
-    "build": "rm -rf lib && yarn run build:esm && yarn run build:cjs && yarn run copy",
+    "build": "rm -rf lib && yarn run build:esm && yarn run build:cjs",
     "build:cjs": "tsc --build tsconfig.compile.json",
     "build:esm": "tsc --build tsconfig.compile.esm.json",
     "test": "cross-env NODE_ENV=test yarn jest",
-    "copy": "cp scripts/backup-index.cjs.js lib/cjs/index.js && cp scripts/backup-index.esm.js lib/esm/index.js  && cp scripts/backup-index.d.ts lib/types/index.d.ts",
-    "prepare": "cp scripts/backup-index.cjs.js lib/cjs/index.js && cp scripts/backup-index.esm.js lib/esm/index.js  && cp scripts/backup-index.d.ts lib/types/index.d.ts",
     "generate": "cd test && prisma -v && prisma generate"
   },
   "private": false,

--- a/packages/orm/prisma/scripts/backup-index.d.ts
+++ b/packages/orm/prisma/scripts/backup-index.d.ts
@@ -1,1 +1,1 @@
-export * from "../.schema";
+export * from "../cjs/.schema";

--- a/packages/orm/prisma/src/cli/generator.ts
+++ b/packages/orm/prisma/src/cli/generator.ts
@@ -1,10 +1,9 @@
 import {generatorHandler} from "@prisma/generator-helper";
-import {generate} from "./prismaGenerator";
-import {join} from "path";
+import {generate, defaultOutput} from "./prismaGenerator";
 
 generatorHandler({
   onManifest: () => ({
-    defaultOutput: join(__dirname, "..", ".schema"),
+    defaultOutput,
     prettyName: "Ts.ED integration",
     requiresGenerators: ["prisma-client-js"]
   }),

--- a/packages/orm/prisma/src/cli/prismaGenerator.ts
+++ b/packages/orm/prisma/src/cli/prismaGenerator.ts
@@ -3,7 +3,10 @@ import {parseEnvValue} from "@prisma/sdk";
 import {promises as asyncFs} from "fs";
 import {generateCode} from "../generator/generateCode";
 import removeDir from "../generator/utils/removeDir";
-import path from "path";
+import path, {join} from "path";
+
+export const defaultOutput = join(__dirname, "..", ".schema");
+export const packageDir = join(__dirname, "..", "..", "..");
 
 function parseStringBoolean(stringBoolean: string | undefined) {
   return Boolean(stringBoolean ? stringBoolean === "true" : undefined);
@@ -27,6 +30,12 @@ export async function generate(options: GeneratorOptions) {
     outputDirPath: outputDir,
     prismaClientPath: prismaClientPath.includes("node_modules") ? "@prisma/client" : toUnixPath(path.relative(outputDir, prismaClientPath))
   });
+
+  if (outputDir === defaultOutput) {
+    await asyncFs.cp(join(packageDir, "scripts", "backup-index.cjs.js"), join(packageDir, "lib", "cjs", "index.js"));
+    await asyncFs.cp(join(packageDir, "scripts", "backup-index.esm.js"), join(packageDir, "lib", "esm", "index.js"));
+    await asyncFs.cp(join(packageDir, "scripts", "backup-index.d.ts"), join(packageDir, "lib", "types", "index.d.ts"));
+  }
 
   return "";
 }

--- a/packages/orm/prisma/src/generator/domain/DmmfModel.ts
+++ b/packages/orm/prisma/src/generator/domain/DmmfModel.ts
@@ -2,6 +2,7 @@ import {DMMF} from "@prisma/generator-helper";
 import {toMap} from "@tsed/core";
 import {ImportDeclarationStructure, StructureKind} from "ts-morph";
 import {DmmfField} from "./DmmfField";
+import {pascalCase} from "change-case";
 
 export class DmmfModel {
   readonly isInputType: boolean;
@@ -61,7 +62,7 @@ export class DmmfModel {
   }
 
   static symbolName(name: string) {
-    return `${name}Model`;
+    return pascalCase(`${name}Model`);
   }
 
   addImportDeclaration(moduleSpecifier: string, name: string, isDefault = false) {

--- a/packages/orm/prisma/src/generator/generateCode.ts
+++ b/packages/orm/prisma/src/generator/generateCode.ts
@@ -1,5 +1,4 @@
 import {DMMF} from "@prisma/generator-helper";
-import path from "path";
 import {CompilerOptions, ModuleKind, Project, ScriptTarget} from "ts-morph";
 import {generateEnums} from "./utils/generateEnums";
 import {generateModels} from "./utils/generateModels";

--- a/packages/orm/prisma/src/generator/utils/generateIndex.ts
+++ b/packages/orm/prisma/src/generator/utils/generateIndex.ts
@@ -7,6 +7,9 @@ export function generateIndex(project: Project, baseDirPath: string, hasEnum: bo
   project.createSourceFile(indexFile, undefined, {overwrite: true}).addExportDeclarations(
     [
       {
+        moduleSpecifier: "./client"
+      },
+      {
         moduleSpecifier: "./interfaces"
       },
       hasEnum && {

--- a/packages/orm/prisma/src/generator/utils/generateRepositories.ts
+++ b/packages/orm/prisma/src/generator/utils/generateRepositories.ts
@@ -45,7 +45,7 @@ export function generateRepositories(dmmf: DMMF.Document, project: Project, base
   const repositoriesIndex = repoDirectory.createSourceFile(`index.ts`, undefined, {overwrite: true});
 
   const exportedModels = models.map((model) => {
-    const name = `${pluralize(model.name)}Repository`;
+    const name = pascalCase(`${pluralize(model.name)}Repository`);
     const modelName = model.toString();
     const sourceFile = repoDirectory.createSourceFile(`${name}.ts`, undefined, {overwrite: true});
 
@@ -191,7 +191,7 @@ export function generateRepositories(dmmf: DMMF.Document, project: Project, base
     addDelegatedMethod({
       repository,
       name: "aggregate",
-      model: model.name
+      model: pascalCase(model.name)
     });
 
     return name;

--- a/packages/orm/prisma/test/prisma/schema.prisma
+++ b/packages/orm/prisma/test/prisma/schema.prisma
@@ -17,7 +17,7 @@ generator tsed {
   emitDMMF                 = true
 }
 
-model User {
+model user {
   /// @TsED.Groups("!creation")
   /// Comment
   id          Int      @id @default(autoincrement())
@@ -29,8 +29,8 @@ model User {
   is18        Boolean?
   name        String?
   successorId Int?
-  successor   User?    @relation("BlogOwnerHistory", fields: [successorId], references: [id])
-  predecessor User?    @relation("BlogOwnerHistory")
+  successor   user?    @relation("BlogOwnerHistory", fields: [successorId], references: [id])
+  predecessor user?    @relation("BlogOwnerHistory")
   role        Role     @default(USER)
   posts       Post[]
   keywords    String[]
@@ -41,7 +41,7 @@ model User {
 
 model Post {
   id     Int   @id @default(autoincrement())
-  user   User? @relation(fields: [userId], references: [id])
+  user   user? @relation(fields: [userId], references: [id])
   userId Int?
 }
 

--- a/packages/orm/prisma/test/snapshots/generate_code/index.ts
+++ b/packages/orm/prisma/test/snapshots/generate_code/index.ts
@@ -1,3 +1,4 @@
+export * from "./client";
 export * from "./interfaces";
 export * from "./enums";
 export * from "./models";

